api/remoterelations/remoterelations.go

```diff
@@ -26,18 +26,146 @@ func NewClient(caller base.APICaller) *Client {
 	return &Client{facadeCaller}
 }
 
+// PublishLocalRelationChange publishes local relations changes to the
+// remote side offering those relations.
+func (c *Client) PublishLocalRelationChange(change params.RemoteRelationsChange) error {
+	args := params.RemoteRelationsChanges{
+		Changes: []params.RemoteRelationsChange{change},
+	}
+	var results params.ErrorResults
+	err := c.facade.FacadeCall("PublishLocalRelationChange", args, &results)
+	if err != nil {
+		return err
+	}
+	if len(results.Results) != 1 {
+		return errors.Errorf("expected 1 result, got %d", len(results.Results))
+	}
+	result := results.Results[0]
+	if result.Error != nil {
+		return result.Error
+	}
+	return nil
+}
+
+// ConsumeRemoteApplicationChange consumes remote changes to applications into the local model.
+func (c *Client) ConsumeRemoteApplicationChange(change params.RemoteApplicationChange) error {
+	args := params.RemoteApplicationChanges{
+		Changes: []params.RemoteApplicationChange{change},
+	}
+	var results params.ErrorResults
+	err := c.facade.FacadeCall("ConsumeRemoteApplicationChange", args, &results)
+	if err != nil {
+		return err
+	}
+	if len(results.Results) != 1 {
+		return errors.Errorf("expected 1 result, got %d", len(results.Results))
+	}
+	result := results.Results[0]
+	if result.Error != nil {
+		return result.Error
+	}
+	return nil
+}
+
+// ExportEntities allocates unique, remote entity IDs for the given entities in the local model.
+func (c *Client) ExportEntities(tags []names.Tag) ([]params.RemoteEntityIdResult, error) {
+	args := params.Entities{Entities: make([]params.Entity, len(tags))}
+	for i, tag := range tags {
+		args.Entities[i].Tag = tag.String()
+	}
+	var results params.RemoteEntityIdResults
+	err := c.facade.FacadeCall("ExportEntities", args, &results)
+	if err != nil {
+		return nil, err
+	}
+	if len(results.Results) != len(tags) {
+		return nil, errors.Errorf("expected %d result(s), got %d", len(tags), len(results.Results))
+	}
+	for _, r := range results.Results {
+		if r.Error != nil {
+			return nil, r.Error
```

Comment:
This approach hides the source of the error (i.e. which entity) and also hides that some may have worked. Is that ok in this case?

Same for similar funcs below.